### PR TITLE
feat(microservices-self-registration): completed discovery server and…

### PR DESCRIPTION
**Pull request by Pat-r1ck**
Microservices Self Registration #2687 

- Microservice structure contains discovery-server & client service
- During startup, client service would register itself with the discovery-server
- During crash, client service would unregister itself from the discovery-server

### CODE REVIEW

**Comment by Pat-r1ck** [comment]
@iluwatar please review my pull request, thank you

File: microservices-self-registration/client-service/pom.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        <version>1.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>client-service</artifactId>
+
+    <properties>
+        <maven.compiler.source>17</maven.compiler.source>
+        <maven.compiler.target>17</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
```
Comment (iluwatar) [inline microservices-self-registration/client-service/pom.xml:18]: Not needed, these come from parent `pom.xml`

File: microservices-self-registration/client-service/src/main/java/com/patrick/clientservice/ClientServiceApplication.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+package com.patrick.clientservice;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.cloud.client.discovery.EnableDiscoveryClient;
+
+@EnableDiscoveryClient
+@SpringBootApplication
+public class ClientServiceApplication
```
Comment (iluwatar) [inline microservices-self-registration/client-service/src/main/java/com/patrick/clientservice/ClientServiceApplication.java:34]: Add JavaDoc to explaing the purpose of this class and its role in the pattern

File: microservices-self-registration/discovery-server/pom.xml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        <version>1.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>discovery-server</artifactId>
+
+    <properties>
+        <maven.compiler.source>17</maven.compiler.source>
+        <maven.compiler.target>17</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
```
Comment (iluwatar) [inline microservices-self-registration/discovery-server/pom.xml:18]: not needed

File: microservices-self-registration/discovery-server/src/main/java/com/patrick/discoveryserver/DiscoveryServerApplication.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+package com.patrick.discoveryserver;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.cloud.netflix.eureka.server.EnableEurekaServer;
+
+@EnableEurekaServer
+@SpringBootApplication
+public class DiscoveryServerApplication
```
Comment (iluwatar) [inline microservices-self-registration/discovery-server/src/main/java/com/patrick/discoveryserver/DiscoveryServerApplication.java:34]: Add JavaDoc to explaing the purpose of this class and its role in the pattern

**Review by iluwatar — CHANGES_REQUESTED** [review]
Additionally, please add this module to the parent `pom.xml`. Otherwise it's not built by CI.

**Pull request closed without merging**